### PR TITLE
Added support for Imperas ISS to uvmt_cv32 UVM testbench for Xcelium

### PR DESCRIPTION
**Pull request by GTumbush**
Shows that ISS .so and UVM .so can both be loaded for Xcelium

### CODE REVIEW

**Comment by GTumbush** [comment]
Hi Mike,
 @eroom1966 was going to delete vendor_lib.rm.  It should be deleted before merging.

Greg

**Comment by MikeOpenHWGroup** [comment]
I can remove the `vendor_lib`.   I note that you have added a clean set of variables in `Common.mk` to point to an OVP installation (see below).   I do not have that one my system, so I will work with @eroom1966 to get that set up.
```
###############################################################################
# Imperas Instruction Set Simulator

#OVPM_DIR   ?= $(PROJ_ROOT_DIR)/vendor_lib/imperas_iss/OVPmodel_encapsulation
#OVP_MODEL_DPI ?= $(OVPM_DIR)/C_OVPmodel/libriscv_sv.Linux64.so

OVP_HOME     ?= $(PROJ_ROOT_DIR)/../riscvImperasDV/imperas
OVP_MODEL_DPI = $(OVP_HOME)/ovp_riscv/obj/Linux64/riscv_CV32E40P.dpi.Linux64.so
OVP_HOME_VLNV = $(OVP_HOME)/ovp_vlnv
OVP_HOME_WRAP = $(OVP_HOME)/ovp_riscv

export IMPERAS_VLNV := ${IMPERAS_VLNV}:$(OVP_HOME_VLNV)/obj

$(OVP_MODEL_DPI):
	#make -C $(OVPM_DIR) compileOVPmodel 
	$(MAKE) -C $(OVP_HOME_VLNV) VLNVSRC=$(OVP_HOME_VLNV) VLNVROOT=$(OVP_HOME_VLNV)/obj 
	$(MAKE) -C $(OVP_HOME_WRAP) SRC=riscv_CV32E40P.dpi.c

c_ovp_model: $(OVP_MODEL_DPI)

clean_c_ovp_model:
	#rm $(OVPM_DIR)/C_OVPmodel/libriscv_sv.*.so
	#rm -rf $(OVPM_DIR)/C_OVPmodel/obj
	$(MAKE) clean -C $(OVP_HOME_VLNV) VLNVSRC=$(OVP_HOME_VLNV) VLNVROOT=$(OVP_HOME_VLNV)/obj 
	$(MAKE) clean -C $(OVP_HOME_WRAP) SRC=riscv_CV32E40P.dpi.c

```
I'll need to get this in place before merging as it break all compiles (except Verilator on the `core` testbench.

**Comment by GTumbush** [comment]
My development environment has iss_integration and riscvImperasDV at the same level.  I clone riscvImperasDV with:
git clone https://github.com/Imperas/riscvImperasDV
git checkout add_riscv_rtl

I doubt you have access to https://github.com/Imperas/riscvImperasDV so perhaps Lee is not done integrating it?

File: cv32/tb/uvmt_cv32/uvmt_cv32_step_compare.sv [hunk 1]
```diff
@@ -273,13 +275,19 @@ module uvmt_cv32_step_compare
 // "xmsim: *W,SLFINV: Call to process::self() from invalid process; returning null"
 final begin
    if (num_pc_checks > 0)
-     $display("step_compare: Checked PC 0d%0d times", num_pc_checks);
+     $display("%m: Checked PC 0d%0d times", num_pc_checks);
```
Comment (MikeOpenHWGroup) [inline cv32/tb/uvmt_cv32/uvmt_cv32_step_compare.sv:278]: Hi @GTumbush .  All of these $display() calls should be `uvm_info() macros.

File: cv32/tb/uvmt_cv32/uvmt_cv32_step_compare.sv [hunk 2]
```diff
@@ -273,13 +275,19 @@ module uvmt_cv32_step_compare
...
    else
-     $display("step_compare: ERROR: PC was checked 0 times!");   
+     $display("%m: ERROR: PC was checked 0 times!");
```
Comment (MikeOpenHWGroup) [inline cv32/tb/uvmt_cv32/uvmt_cv32_step_compare.sv:280]: This must be replaced with `uvm_error().

**Comment by GTumbush** [comment]
I started with uvm_info and uvm_error but Xcelium flagged many "xmsim: *W,SLFINV: Call to process::self() from invalid process; returning null" warnings.  Also, these run after the UVM_ERROR and UVM_WARNING reports so will not add to this report. 

Open to ideas on how to get this check called before the UVM_ERROR and UVM_WARNING reports.

**Comment by MikeOpenHWGroup** [comment]
Ah, yes.  The checks are in a final block - should have noticed that.  We already have one final block in the testbench (uvmt_cv32_tb) and its never good to have two.

The "proper" way to do this would be to implement step-and-compare as a uvm_component and use explicit phasing.  Another way to do it would be to pass the vp_status_interface down to step and compare and use (test_pass||test_fail||exit_valid) to know when the test-program thinks its done.

This is getting into the guts of the environment, so leave that with me.

**Comment by GTumbush** [comment]
Hi Mike, I can handle changing the step_compare into a class and using the phasing.  OK if I do it?
Greg

**Comment by MikeOpenHWGroup** [comment]
Absolutely!

---mike

**Comment by GTumbush** [comment]
The bind method was getting messy and hard to follow. Ended up calling a task from uvmt_cv32_base_test::phase_ended to display step/compare check counts.
